webpack/components/extensions/HostDetails/Tabs/EnableTracerModal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Flex,
+  FlexItem,
+  Modal,
+  ModalVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+} from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { useSelector, useDispatch } from 'react-redux';
+import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
+import { installTracerPackage } from './HostTracesActions';
+import { KATELLO_TRACER_PACKAGE } from './HostTracesConstants';
+import { REX_FEATURES } from './RemoteExecutionConstants';
+
+const EnableTracerModal = ({ isOpen, setIsOpen }) => {
+  const title = __('Enable Tracer');
+  const body = __('Enabling will install the katello-host-tools-tracer package on the host.');
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const toggleDropdownOpen = () => setIsDropdownOpen(prev => !prev);
+  const dropdownOptions = [
+    __('via remote execution'),
+    __('via customized remote execution'),
+  ];
+  const [selectedOption, setSelectedOption] = useState(dropdownOptions[0]);
+  const dispatch = useDispatch();
+  const hostDetails = useSelector(state => selectAPIResponse(state, 'HOST_DETAILS'));
+  const { name: hostname } = hostDetails;
+  const handleSelect = () => {
+    setIsDropdownOpen(false);
+  };
+  const enableTracer = () => {
+    dispatch(installTracerPackage({ hostname }));
+    setIsOpen(false);
+  };
+
+  const dropdownItems = dropdownOptions.map(text => (
+    <DropdownItem key={`option_${text}`} onClick={() => setSelectedOption(text)}>{text}</DropdownItem>
+  ));
+
+  const customizedRexUrl = () => {
+    const urlQuery = encodeURI([
+      `feature=${REX_FEATURES.KATELLO_PACKAGE_INSTALL}`,
+      `inputs[package]=${KATELLO_TRACER_PACKAGE}`,
+      `host_ids=name ^ (${hostname})`,
+    ].join('&'));
+    return `/job_invocations/new?${urlQuery}`;
+  };
+
+  const getEnableTracerButton = () => {
+    const [viaRex] = dropdownOptions;
+    if (selectedOption === viaRex) {
+      return (
+        <Button
+          key="enable_button"
+          type="submit"
+          variant="primary"
+          onClick={enableTracer}
+        >
+          {title}
+        </Button>
+      );
+    }
+    return (
+      <Button
+        key="enable_button"
+        component="a"
+        isLoading={redirect}
+        onClick={() => setRedirect(true)}
```

Comment:
Don't know enough to test this but what is `redirect` doing here other than the isLoading bit?

Reply:
oh nvm..It's a href button so I am assuming it acts as a link..

Reply:
I was just using it to put the spinner on the button when you click it.  It was badly named.  Updated the name to `setButtonLoading`. :+1: